autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/LibLinearSVCPreprocessor.py

```diff
@@ -0,0 +1,141 @@
+                 intercept_scaling: int = 1, fit_intercept: bool = True,
+                 random_state: Optional[np.random.RandomState] = None):
+
+        self.dual = dual
+        self.penalty = penalty
+        self.loss = loss
+        self.multi_class = multi_class
+        self.intercept_scaling = intercept_scaling
+        self.fit_intercept = fit_intercept
+        self.tol = tol
+        self.C = C
+
+        super().__init__(random_state=random_state)
+
+    def get_components_kwargs(self) -> Dict[str, Any]:
+        """
+        returns keyword arguments required by the feature preprocessor
+
+        Returns:
+            Dict[str, Any]: kwargs
+        """
+        return dict(
+            dual=self.dual,
+            penalty=self.penalty,
+            loss=self.loss,
+            multi_class=self.multi_class,
+            intercept_scaling=self.intercept_scaling,
+            tol=self.tol,
+            fit_intercept=self.fit_intercept,
+            C=self.C,
+            random_state=self.random_state
+        )
+
+    def fit(self, X: Dict[str, Any], y: Any = None) -> BaseEstimator:
+
+        # TODO: add class_weights
+        estimator = LinearSVC(**self.get_components_kwargs())
+
+        self.preprocessor['numerical'] = SelectFromModel(estimator=estimator,
+                                                         threshold='mean',
+                                                         prefit=False)
+        return self
+
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, BaseDatasetPropertiesType]] = None) -> Dict[str, Any]:
+        return {'shortname': 'LinearSVC Preprocessor',
+                'name': 'Liblinear Support Vector Classification Preprocessing',
+                'handles_sparse': True,
+                'handles_classification': True,
+                'handles_regression': False
+                }
+
+    @staticmethod
+    def get_hyperparameter_search_space(
+        dataset_properties: Optional[Dict[str, BaseDatasetPropertiesType]] = None,
+        penalty: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='penalty',
+                                                                       value_range=("l1",),
+                                                                       default_value="l1",
+                                                                       ),
+        loss: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='loss',
+                                                                    value_range=("squared_hinge", "hinge"),
+                                                                    default_value="squared_hinge",
+                                                                    ),
+        dual: HyperparameterSearchSpace = HyperparameterSearchSpace(hyperparameter='dual',
+                                                                    value_range=(False,),
```

Comment:
constant?

Reply:
yes as default they are constant. But having it this way allows the user to pass search space updates to also search for these hyperparameters